### PR TITLE
OUT-65 | Save settings based on Workspace ID

### PR DESCRIPTION
**Pull request by rrojan**
Ref: OUT-65

Changes:

- Make senderId non-nullable in Settings
- Add workspaceId field in Settings table
- Save and retrieve settings based on workspaceId instead of createdById
- Remove code related to fetching settings with createdById

### CODE REVIEW

File: migrations/20240216014724_remove_uuid_constraint_in_settings/migration.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+-- AlterTable
+ALTER TABLE "Setting" ALTER COLUMN "workspaceId" SET DATA TYPE TEXT;
```
Comment (sajxraj) [inline migrations/20240216014724_remove_uuid_constraint_in_settings/migration.sql:2]: workspace id should be a uuid. Any reason for making it text?

File: src/app/page.tsx [hunk 2]
```diff
@@ -58,7 +59,7 @@ async function getInternalUsers(token: string): Promise<InternalUsers> {
 }
 
 export default async function Page({ searchParams }: { searchParams: SearchParams }) {
-  const { me } = await getContent(searchParams);
+  const { me, workspace } = await getContent(searchParams);
```
Comment (sajxraj) [inline src/app/page.tsx:62]: will there be workspace in the url query param?

File: src/types/common.ts [hunk 3]
```diff
@@ -9,6 +9,27 @@ export const MeResponseSchema = z.object({
...
+  squareIconUrl: z.string(),
+  fullLogoUrl: z.string(),
+  squareLoginImageUrl: z.string(),
+  socialSharingImageUrl: z.string(),
+  colorSidebarBackground: z.string(),
+  colorSidebarText: z.string(),
+  colorAccent: z.string(),
+  font: z.string(),
+  metaTitle: z.string(),
+  metaDescription: z.string(),
```
Comment (sajxraj) [inline src/types/common.ts]: so none of these are optional/nullable fields? Please test in workspace if any of these fields may not come.

File: src/utils/common.ts [hunk 4]
```diff
@@ -14,6 +14,11 @@ export function errorHandler(message: string, status: number = 200) {
   );
 }
 
+export async function getWorkspace(apiToken: string) {
```
Comment (sajxraj) [inline src/utils/common.ts]: return type missing

File: src/utils/copilotApiUtils.ts [hunk 5]
```diff
@@ -28,6 +30,10 @@ export class CopilotAPI {
     return MeResponseSchema.parse(await this.copilot.getUserInfo());
   }
 
+  async getWorkspace(): Promise<WorkspaceResponse> {
+    return await WorkspaceResponseSchema.parse(await this.copilot.getWorkspaceInfo());
```
Comment (sajxraj) [inline src/utils/copilotApiUtils.ts]:
```suggestion
    return WorkspaceResponseSchema.parse(await this.copilot.getWorkspaceInfo());
```

File: migrations/20240216014724_remove_uuid_constraint_in_settings/migration.sql (hunk 1 above)
Comment (rrojan) [reply migrations/20240216014724_remove_uuid_constraint_in_settings/migration.sql:2]: `workspaceId` is currently being decrypted from token payload in this format:

image

File: src/types/common.ts (hunk 3 above)
Comment (rrojan) [reply src/types/common.ts]: Fixed according to type definition in SDK
image

**Review by sajxraj — APPROVED** [review]

**Pull request merged**